### PR TITLE
Simplifying some code in framework type code style

### PR DESCRIPTION
**Pull request by balajikris**
Addressing some last pieces of feedback from https://github.com/dotnet/roslyn/pull/13132

1. Simplify some code, in general
2. Check options ahead of time before performing any analysis and bail early. 

No functional change, hence no new tests required.

### CODE REVIEW

**Comment by balajikris** [comment]
@dotnet/roslyn-ide please review.

**Comment by balajikris** [comment]
ping, @dotnet/roslyn-ide

File: src/Features/Core/Portable/Diagnostics/Analyzers/PreferFrameworkTypeDiagnosticAnalyzerBase.cs [hunk 1]
```diff
@@ -70,45 +88,70 @@ protected void AnalyzeNode(SyntaxNodeAnalysisContext context)
...
+            var optionValue = optionSet.GetOption(option, language);
+            return OptionSettingPrefersFrameworkType(optionValue, optionValue.Notification.Value);
         }
+
+        /// <summary>
+        /// checks if style is preferred and the enforcement is not None.
+        /// </summary>
+        /// <remarks>if predefined type is not preferred, it implies the preference is framework type.</remarks>
+        private static bool OptionSettingPrefersFrameworkType(CodeStyleOption<bool> optionValue, DiagnosticSeverity severity) =>
+            !optionValue.Value && severity != DiagnosticSeverity.Hidden;
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/Diagnostics/Analyzers/PreferFrameworkTypeDiagnosticAnalyzerBase.cs]: i don't quite get the !optionValue part.

Comment (balajikris) [inline src/Features/Core/Portable/Diagnostics/Analyzers/PreferFrameworkTypeDiagnosticAnalyzerBase.cs]: Hmm. Basically these are boolean options - true implies preference towards predefined type, false implies framework type.  Which is what i tried to explain in hte `remarks` tag.

**Comment by CyrusNajmabadi** [comment]
:+1:

File: src/Features/Core/Portable/Diagnostics/Analyzers/PreferFrameworkTypeDiagnosticAnalyzerBase.cs [hunk 2]
```diff
@@ -70,45 +88,70 @@ protected void AnalyzeNode(SyntaxNodeAnalysisContext context)
...
-                context.ReportDiagnostic(Diagnostic.Create(descriptor, predefinedTypeNode.GetLocation()));
-            }
+        private bool IsFrameworkTypePreferred(OptionSet optionSet, PerLanguageOption<CodeStyleOption<bool>> option, string language)
+        {
+            var optionValue = optionSet.GetOption(option, language);
+            return OptionSettingPrefersFrameworkType(optionValue, optionValue.Notification.Value);
         }
+
+        /// <summary>
+        /// checks if style is preferred and the enforcement is not None.
```
Comment (Pilchie) [inline src/Features/Core/Portable/Diagnostics/Analyzers/PreferFrameworkTypeDiagnosticAnalyzerBase.cs]: Nit: Doc comments should be sentences (start with a capital).

File: src/Features/Core/Portable/Diagnostics/Analyzers/PreferFrameworkTypeDiagnosticAnalyzerBase.cs (hunk 1 above)
Comment (Pilchie) [inline src/Features/Core/Portable/Diagnostics/Analyzers/PreferFrameworkTypeDiagnosticAnalyzerBase.cs]: I would make this `<returns>True if the predefined type keyword is preferred, false if the framework type name is preferred</returns>`

**Comment by Pilchie** [comment]
Comment nits, but :+1:

**Pull request merged**